gdal/swig/python/osgeo/gdal.py

```diff
@@ -3141,7 +3141,7 @@ def ReadAsMaskedArray(self,
         if mask is not None:
             array = self.ReadAsArray(array_start_idx, count, array_step)
             mask_array = mask.ReadAsArray(array_start_idx, count, array_step)
-            bool_array = ~mask_array.astype(numpy.bool)
+            bool_array = ~mask_array.astype(bool)
```

Comment:
The related changed in the source file swig/include/python/gdal_python.i should also be committed